protocol/storage/miner.go

```diff
@@ -706,70 +700,61 @@ func (sm *Miner) OnNewHeaviestTipSet(ts types.TipSet) {
 		return
 	}
 
-	provingPeriodStart, err := sm.getProvingPeriodStart()
+	provingPeriodStart, provingPeriodEnd, err := sm.getProvingPeriod()
 	if err != nil {
-		log.Errorf("failed to get provingPeriodStart: %s", err)
+		sm.log.Errorf("failed to get proving period: %s", err)
 		return
 	}
 
 	sm.postInProcessLk.Lock()
 	defer sm.postInProcessLk.Unlock()
 
-	if sm.postInProcess != nil && sm.postInProcess.Equal(provingPeriodStart) {
+	if sm.postInProcess != nil && sm.postInProcess.Equal(provingPeriodEnd) {
 		// post is already being generated for this period, nothing to do
 		return
 	}
 
 	height, err := ts.Height()
 	if err != nil {
-		log.Errorf("failed to get block height: %s", err)
-		return
-	}
-
-	sectorSize, err := sm.porcelainAPI.MinerGetSectorSize(ctx, sm.minerAddr)
-	if err != nil {
-		log.Errorf("failed to get miner's sector size: %s", err)
+		sm.log.Errorf("failed to get block height: %s", err)
 		return
 	}
 
 	// the block height of the new heaviest tipset
 	h := types.NewBlockHeight(height)
 
-	// compute the block height at which the miner's current proving period ends
-	provingPeriodEnd := provingPeriodStart.Add(types.NewBlockHeight(miner.ProvingPeriodDuration(sectorSize)))
-
 	if h.GreaterEqual(provingPeriodStart) {
 		if h.LessThan(provingPeriodEnd) {
 			// we are in a new proving period, lets get this post going
-			sm.postInProcess = provingPeriodStart
+			sm.postInProcess = provingPeriodEnd
 
-			seed, err := sm.currentProvingPeriodPoStChallengeSeed(ctx)
+			seed, err := sm.currentProvingPeriodPoStChallengeSeed(ctx, provingPeriodStart)
 			if err != nil {
-				log.Errorf("error obtaining challenge seed: %s", err)
+				sm.log.Errorf("error obtaining challenge seed: %s", err)
 				return
 			}
 
 			go sm.submitPoSt(provingPeriodStart, provingPeriodEnd, seed, inputs)
 		} else {
 			// we are too late
 			// TODO: figure out faults and payments here
-			log.Errorf("too late start=%s  end=%s current=%s", provingPeriodStart, provingPeriodEnd, h)
+			sm.log.Errorf("too late start=%s  end=%s current=%s", provingPeriodEnd, provingPeriodEnd, h)
```

Comment:
BLOCKING: The first non-format-string argument should be `provingPeriodStart`, not `provingPeriodEnd`.